tasks/src/tasks/store.rs

```diff
@@ -0,0 +1,110 @@
+use std::{collections::HashMap, num::NonZeroU32};
+
+use imap_types::{
+    command::CommandBody,
+    core::Vec1,
+    fetch::MessageDataItem,
+    flag::{Flag, StoreResponse, StoreType},
+    response::{Data, StatusBody, StatusKind},
+    sequence::SequenceSet,
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug)]
+pub struct StoreTask {
+    sequence_set: SequenceSet,
+    kind: StoreType,
+    flags: Vec<Flag<'static>>,
+    uid: bool,
+    output: HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>,
+}
+
+impl Task for StoreTask {
+    type Output = Result<HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Store {
+            sequence_set: self.sequence_set.clone(),
+            kind: self.kind,
+            response: StoreResponse::Answer,
+            flags: self.flags.clone(),
+            uid: self.uid,
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        if let Data::Fetch { items, seq } = data {
+            self.output.insert(seq, items);
+            None
+        } else {
+            Some(data)
+        }
+    }
+
+    fn process_tagged(self, status_body: StatusBody<'static>) -> Self::Output {
+        match status_body.kind {
+            StatusKind::Ok => Ok(self.output),
+            StatusKind::No => Err(TaskError::UnexpectedNoResponse(status_body)),
```

Comment:
Just to be clear: The `Code` can potentially also be used here...

Reply:
But in this case it is already exposed? I mean, if a client wants to display information from `Code` or `Text`, it has to pattern match the `Result` and check by itself the `status_body`?

Reply:
Ah, yes. This was just a reminder for us that errors also potentially have codes. All good here :-)